assets/common.sh

```diff
@@ -260,3 +260,39 @@ load_git_crypt_key() {
       cat $git_crypt_tmp_key_path | tr ' ' '\n' | base64 -d > $GIT_CRYPT_KEY_PATH
   fi
 }
+
+setup_github_app_credentials() {
+  github_app_id=$(jq -r '.source.github_app_id // ""' <<< "$payload")
+  github_app_private_key=$(jq -r '.source.github_app_private_key // ""' <<< "$payload")
+  github_app_installation_id=$(jq -r '.source.github_app_installation_id // ""' <<< "$payload")
+
+  if [ -n "$github_app_id" ] && [ -n "$github_app_private_key" ] && [ -n "$github_app_installation_id" ]; then
+    # Generate JWT for GitHub App
+    now=$(date +%s)
+    iat=$((now - 60))
+    exp=$((now + 600))  # 10 minutes expiration
+    jwt_header=$(echo -n '{"alg":"RS256","typ":"JWT"}' | base64 | tr -d '=' | tr '/+' '_-' | tr -d '\r\n')
+    jwt_payload=$(echo -n "{\"iat\":$iat,\"exp\":$exp,\"iss\":\"$github_app_id\"}" | base64 | tr -d '=' | tr '/+' '_-' | tr -d '\r\n')
+    jwt_signature=$(echo -n "$jwt_header.$jwt_payload" | openssl dgst -sha256 -sign <(echo -n "$github_app_private_key") | base64 | tr -d '=' | tr '/+' '_-' | tr -d '\r\n')
+    jwt_token="$jwt_header.$jwt_payload.$jwt_signature"
+
+    # Get installation access token
+    access_token=$(curl -s -X POST -H "Authorization: Bearer $jwt_token" -H "Accept: application/vnd.github.v3+json" \
+      "https://api.github.com/app/installations/$github_app_installation_id/access_tokens" | jq -r '.token')
```

Comment:
I'm guessing this `curl` call is what stopped you from directly testing this function and resulted in the copy-pasted tests.

We can do two things here to actually test this curl call:
1. Make the api endpoint `https://api.github.com` configurable
2. Use netcat `nc` to mock the server response

Basic example of using netcat to mock a server response:
```
echo -e "HTTP/1.1 200 OK\r\nContent-Type: application/json\r\n\r\n{\"token\":\"some-token\"}" | nc -l 8080 &
curl http://localhost:8080
```

---

Another idea! In the test suite, you could add a `curl()` function that gets called instead of real `curl`.

```bash
curl() {
  // test and check passed in args
}

export -f curl
```

There are options, we can get this code properly tested :)